app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java

```diff
@@ -0,0 +1,148 @@
+        this.studyId = studyId;
+        this.zone = zone;
+        this.endsOn = endsOn;
+        this.healthCode = healthCode;
+        this.events = (events == null) ? null : ImmutableMap.copyOf(events);
+        this.schedulePlanGuid = schedulePlanGuid; 
+    }
+    
+    /**
+     * Copy method to copy existing values, augmenting the context with an event map.
+     * @param events
+     * @return
+     */
+    public ScheduleContext withEvents(Map<String,DateTime> events) {
+        return new ScheduleContext(this.studyId, this.zone, this.endsOn, this.healthCode, events, this.schedulePlanGuid);
+    }
+    
+    /**
+     * Copy method to copy existing values, augmenting the context with a schedule plan GUID.
+     * @param schedulePlanGuid
+     * @return
+     */
+    public ScheduleContext withSchedulePlan(String schedulePlanGuid) {
+        return new ScheduleContext(this.studyId, this.zone, this.endsOn, this.healthCode, this.events, schedulePlanGuid);
+    }
+    
+    /**
+     * The study identifier for this participant.
+     * @return
+     */
+    public StudyIdentifier getStudyIdentifier() {
+        return studyId;
+    }
+    
+    /**
+     * The time zone of the client at the time of this request. This allows the scheduler to accommodate 
+     * moves between time zones.
+     * @return
+     */
+    public DateTimeZone getZone() {
+        return zone;
+    }
+    
+    /**
+     * The current request is asking for tasks up to a given end date.
+     * @return
+     */
+    public DateTime getEndsOn() {
+        return endsOn;
+    }
+
+    /**
+     * The current user's health code.
+     * @return
+     */
+    public String getHealthCode() {
+        return healthCode;
+    }
+    
+    /**
+     * The schedule plan providing the schedule, used to keep track of individual runs of the 
+     * scheduler to generate a set of tasks.
+     * @return
+     */
+    public String getSchedulePlanGuid() {
+        return schedulePlanGuid;
+    }
+    
+    /**
+     * Are there any events recorded for this participant? This should always return true since every 
+     * participant should have an enrollment event, if nothing else.
+     * @return
+     */
+    public boolean hasEvents() {
+        return (events != null && !events.isEmpty());
+    }
+    
+    /**
+     * Get an event timestamp for a given event ID.
+     * @param eventId
+     * @return
+     */
+    public DateTime getEvent(String eventId) {
+        return (events != null) ? events.get(eventId) : null;
+    }
+    
+    /**
+     * Returns now in the user's time zone. Practically this is not that important but 
+     * it allows you to calculate all time calculations in one time zone, which is easier 
+     * to reason about.
+     * @return
+     */
+    public DateTime getNow() {
+        return DateTime.now(zone);
+    }
```

Comment:
Is this necessary? DateTimes represent a complete timestamp. isBefore()/isAfter() work just fine regardless of time zone.